src/MultiFormats/NetworkProtocol.cs

```diff
@@ -0,0 +1,557 @@
+        public override uint Code => 443;
+    }
+
+    internal class WsNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "ws";
+        public override uint Code => 477;
+    }
+
+    internal class WssNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "wss";
+        public override uint Code => 478;
+    }
+
+    internal class Libp2PWebrtcStarNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "libp2p-webrtc-star";
+        public override uint Code => 275;
+    }
+
+    internal class Libp2PWebrtcDirectNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "libp2p-webrtc-direct";
+        public override uint Code => 276;
+    }
+
+    internal class UdtNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "udt";
+        public override uint Code => 301;
+    }
+
+    internal class UtpNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "utp";
+        public override uint Code => 302;
+    }
+
+    internal class P2PCircuitNetworkProtocol : ValuelessNetworkProtocol
+    {
+        public override string Name => "p2p-circuit";
+        public override uint Code => 290;
+    }
+
+    internal abstract class DomainNameNetworkProtocol : NetworkProtocol
+    {
+        public string DomainName { get; set; }
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            DomainName = Value;
+        }
+
+        public override void ReadValue(CodedInputStream stream)
+        {
+            Value = stream.ReadString();
+            DomainName = Value;
+        }
+
+        public override void WriteValue(TextWriter stream)
+        {
+            stream.Write('/');
+            stream.Write(DomainName.ToString());
+        }
+
+        public override void WriteValue(CodedOutputStream stream) { stream.WriteString(DomainName); }
+    }
+
+    internal class DnsNetworkProtocol : DomainNameNetworkProtocol
+    {
+        public override string Name => "dns";
+        public override uint Code => 53;
+    }
+
+    internal class DnsAddrNetworkProtocol : DomainNameNetworkProtocol
+    {
+        public override string Name => "dnsaddr";
+        public override uint Code => 56;
+    }
+
+    internal class Dns4NetworkProtocol : DomainNameNetworkProtocol
+    {
+        public override string Name => "dns4";
+        public override uint Code => 54;
+    }
+
+    internal class Dns6NetworkProtocol : DomainNameNetworkProtocol
+    {
+        public override string Name => "dns6";
+        public override uint Code => 55;
+    }
+
+    internal class IpcidrNetworkProtocol : NetworkProtocol
+    {
+        public ushort RoutingPrefix { get; set; }
+
+        public override string Name => "ipcidr";
+
+        // TODO: https://github.com/multiformats/multiaddr/issues/60
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)